### DISCUSSION TITLE
Header evolution

### DISCUSSION THREAD

**Discussion by yceballost** [Telefonica/mistica-design#493] [Proposals]
![image](https://user-images.githubusercontent.com/6722153/175555646-eae4e316-b931-4df1-8f48-40d32eba8bce.png)

Actualmente un header tiene los elementos de la parte superior, creo que esto componente nació con demasiados elementos pensados para casos muy concretos, como lo son ese Pretitle + amount + buttons

Creo que este componente debería evolucionar a tener un pretitle, title y subtitle, y el resto debería ser un slot.

![image](https://user-images.githubusercontent.com/6722153/175555985-c9238f44-ae36-4902-9b7d-77695136a292.png)

**Comment by yceballost** [comment]
Algo que tendríamos que tener en cuenta aquí es que actualmente los headers tiene sus propios padding laterales, por lo que con este planteamiento no sería óptimo usar un carousel dentro del Slot del Header. (aunque se podría solucionar con un NegativeBox)

**Comment by yceballost** [comment]
Añado una info que estaba en un ticket de Jira, a valorar

For the web context inside Account use cases, we need some changes.

## HeaderLayout
extra content must remain in vertical on desktop viewports, except we optionally ask to move it to the right.

## Header
- Support for 2 buttons: we already have cases with 2 Call To Action.
- Title must be optional: we have valid use cases where the title content isn't provided by the OB.
- Header must accept any kind of content (aka React.Node) to support skeletons.

**Comment by yceballost** [comment]
Valorar el uso actual de este componente para mantener esas props muy custom de un producto Novum y evitar un gran breaking change.

**Comment by sergivillar** [comment]
En zeus web, necesitariamos añadir un Avatar a la izquierda, y seguir teniendo el campo extra para el selector de Facilities.

https://www.figma.com/file/cq11QvfivKdK6JycYlNgtO/Consumos-Main?node-id=5882%3A34899
